macOSX / clang compile fails with undefined locale identifiers

macOSX build now failing with undefined locale identifiiers, after merge of #7997 :

```
clang: numpy/core/src/multiarray/numpyos.c
numpy/core/src/multiarray/numpyos.c:518:5: error: use of undeclared identifier 'locale_t'
    locale_t clocale;
```

https://travis-ci.org/MacPython/numpy-wheels/jobs/205229905#L3775

Errors result from merge of https://github.com/numpy/numpy/pull/7997 - commit 23ac8554b builds (pre-merge), commit 7bc881d05 breaks (the merge commit).

@juliantaylor - any thoughts?

--- a/numpy/core/setup_common.py
+++ b/numpy/core/setup_common.py
@@ -116,7 +116,7 @@ OPTIONAL_HEADERS = [
                 "xmmintrin.h",  # SSE
                 "emmintrin.h",  # SSE2
                 "features.h",  # for glibc version linux
-                "xlocale.h"  # see GH#8367
+                "xlocale.h",  # see GH#8367
                 "dlfcn.h", # dladdr
 ]
 

--- a/numpy/core/src/multiarray/temp_elide.c
+++ b/numpy/core/src/multiarray/temp_elide.c
@@ -3,6 +3,7 @@
 
 #define NPY_NO_DEPRECATED_API NPY_API_VERSION
 #define _MULTIARRAYMODULE
+#include "npy_config.h"
 #include "numpy/arrayobject.h"
 
 #define NPY_NUMBER_MAX(a, b) ((a) > (b) ? (a) : (b))

